llvm/lib/CodeGen/MachineScheduler.cpp

```diff
@@ -4266,6 +4266,12 @@ unsigned ResourceSegments::getFirstAvailableAt(
   assert(std::is_sorted(std::begin(_Intervals), std::end(_Intervals),
                         sortIntervals) &&
          "Cannot execute on an un-sorted set of intervals.");
+
+  // Zero resource usage is allowed by TargetSchedule.td but we do not construct
+  // a ResourceSegment interval for that situation.
+  if (AcquireAtCycle == Cycle)
```

Comment:
```
llvm-project/llvm/lib/CodeGen/MachineScheduler.cpp:4281:25: error: use of undeclared identifier 'Cycle'
  if (AcquireAtCycle == Cycle)
                        ^
```

https://lab.llvm.org/buildbot/#/builders/225/builds/30132

Reply:
weird, isn't `Cycle` the function parameter? Did a rebase get messed up when landing?

Reply:
I committed a rename from Cycle -> ReleaseAtCycle which included the name of the arg in the function. I forgot to rebase this patch on top of that and I introduced a use of `Cycle` in this patch.